src/logic/single_player_game_settings_provider.cc

```diff
@@ -92,10 +92,10 @@ void SinglePlayerGameSettingsProvider::set_map(const std::string& mapname,
 	s.mapname = mapname;
 	s.mapfilename = mapfilename;
 	s.savegame = savegame;
-
-	uint32_t oldplayers = s.players.size();
 	s.players.resize(maxplayers);
 
+	set_player_number(0);
+	uint32_t oldplayers = 0;  // s.players.size();
```

Comment:
What do you mean by this?

Reply:
deactivated old behaviour...
I've just rephrased it slightly to  emphasize map settings are reset when map is changed